addons/html_editor/static/src/plugin.js

```diff
@@ -111,11 +111,19 @@ export class Plugin {
     addDomListener(target, eventName, fn, capture) {
         const handler = fn.bind(this);
         target.addEventListener(eventName, handler, capture);
-        this._cleanups.push(() => target.removeEventListener(eventName, handler, capture));
+        const hash = eventName + (Math.random() + 1).toString(36).substring(7);
+        this._cleanups[hash] = () => target.removeEventListener(eventName, handler, capture);
+        return hash;
+    }
+    removeDomListener(hash) {
+        if (this._cleanups[hash]) {
+            this._cleanups[hash]();
+            delete this._cleanups[hash];
+        }
```

Comment:
If you really want to be able to add the method removeDomListener, I would rather use the reference of the method rather than a generated hash.

```js
this._cleanups = new Map();
// ...
addDomListener(target, eventName, fn, capture) {
    const handler = fn.bind(this);
    target.addEventListener(eventName, handler, capture);
    this._cleanups.set(fn, () => target.removeEventListener(eventName, handler, capture));
}
removeDomListener(fn) {
  this._cleanups.delete(fn);
}
```